src/geff/_graph_libs/_api_wrapper.py

```diff
@@ -18,137 +16,125 @@
     from geff.metadata._schema import GeffMetadata
     from geff.validate.data import ValidationConfig
 
-SupportedBackend = Literal["networkx", "rustworkx", "spatial-graph"]
-
+    from ._backend_protocol import Backend
 
-class ConstructFunc(Protocol[R]):
-    """A protocol for callables that construct a graph from GEFF data."""
-
-    def __call__(
-        self,
-        metadata: GeffMetadata,
-        node_ids: NDArray[Any],
-        edge_ids: NDArray[Any],
-        node_props: dict[str, PropDictNpArray],
-        edge_props: dict[str, PropDictNpArray],
-        *args: Any,
-        **kwargs: Any,
-    ) -> R:
-        """
-        The callable must have this function signature.
-
-        The callable must have the first argument `in_memory_geff`, it may have additional
-        args and kwargs.
-
-        Args:
-            metadata (GeffMetadata): The metadata of the graph.
-            node_ids (np.ndarray): An array containing the node ids. Must have same dtype as
-                edge_ids.
-            edge_ids (np.ndarray): An array containing the edge ids. Must have same dtype
-                as node_ids.
-            node_props (dict[str, tuple[np.ndarray, np.ndarray | None]] | None): A dictionary
-                from node property names to (values, missing) arrays, which should have same
-                length as node_ids.
-            edge_props (dict[str, tuple[np.ndarray, np.ndarray | None]] | None): A dictionary
-                from edge property names to (values, missing) arrays, which should have same
-                length as edge_ids.
-            *args (Any): Optional args for constructing the `in_memory_geff`.
-            **kwargs (Any): Optional kwargs for constructing the `in_memory_geff`.
-
-        Returns:
-            A graph object instance for a particular backend.
-        """
-        ...
+    NxGraph: TypeAlias = nx.Graph | nx.DiGraph
+    RxGraph: TypeAlias = rx.PyGraph | rx.PyDiGraph
+    SgGraph: TypeAlias = sg.SpatialGraph | sg.SpatialDiGraph
+    SupportedGraphType: TypeAlias = NxGraph | RxGraph | SgGraph
```

Comment:
just noting ... (but i don't think anything needs to be done here... this PR has been open long enough and is a very nice improvement already):

the `get_backend_from_graph_type`, is the only thing that requires these SupportedGraphTypes, and is the only remaining thing in this `_api_wrapper` that requires backend-specific knowledge (i.e. where the abstraction leaks a bit). 

One could have a `def supports()` method on the backends, so rather than *you* doing the loop with 

```
    for backend, graph_type in BACKEND_TYPE_MAP.items():
        if isinstance(graph, graph_type):
            return get_backend(backend)
```

it would be `for backend in backends: if backend.supports(graph)`...  and then all the `BACKEND_TYPE_MAP` and other logic gets removed here.